File: mongo-target/src/main/java/com/japca/mongo/MongoController.java
package com.japca.mongo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by Jakub Krhovják on 5/10/18.
 */



@RestController
public class MongoController {

	@Autowired
	private MongoDao mongoDao;

	@PostMapping("/item")
	public void saveItem(@RequestBody Record record) {
		mongoDao.save(record);
	}

	@GetMapping("/item/{id}")
	public Record getItem(@PathVariable("id") Long id) {
         return mongoDao.findByIdentifier(id);
	}

}

File: mongo-target/src/main/kotlin/com/japca/mongo/MongoController.kt
package com.japca.mongo

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

/**
 * Created by Jakub Krhovják on 5/10/18.
 */


@RestController
class MongoController {

    @Autowired
    lateinit var mongoDao: MongoDao

    @PostMapping("/record")
    fun saveRecord(@RequestBody record: Record): Mono<Record> {
        return mongoDao.save(record)
    }

    @GetMapping("/record/{id}")
    fun getRecord(@PathVariable("id") id: Long): Flux<Record> {
        return mongoDao.findByIdentifier(id)
    }

}
